cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MessagingMetricsProcessorService.java

```diff
@@ -380,6 +395,19 @@ public void run() {
           if (sleepTime > 0) {
             TimeUnit.MILLISECONDS.sleep(sleepTime);
           }
+
+          // shut down the executores if they are not needed anymore
+          // todo uncomment and run only every 100 times
+          /*if (instanceId == 0 && topicIdMetaKey.getTopicId().equals(metricsTopics.get(0))) {
+            if (dataMigrator != null && dataMigratorExecutor != null && !dataMigratorExecutor.isShutdown()
+              && dataMigrator.isMigrationComplete()) {
```

Comment:
It seems that that would be ok: https://stackoverflow.com/questions/29698934/shutting-down-an-executorservice-from-running-task